File: 09-05-2026/clear_digits.py
def clearDigits(s):
    chars = []

    for c in s:
        if not c.isdigit():
            chars.append(c)
        else:
            chars.pop()

    return ''.join(chars)

print(clearDigits("abc"))  # abc
print(clearDigits("a1b2c3"))  # ''
print(clearDigits("cba34"))  # 'c'

File: 09-05-2026/clear_digits.js
const clearDigits = (s) => {
  const chars = [];

  for (let i = 0; i < s.length; i++) {
    if (isNaN(s[i])) chars.push(s[i]);
    else chars.pop();
  }
  return chars.join("");
};

console.log(clearDigits("abc")); // abc
console.log(clearDigits("a1b2c3")); // ''
console.log(clearDigits("cba34")); // 'c'
